src/error.rs

```diff
@@ -85,7 +85,7 @@ impl Error {
     }
 
     // backcompat shim
-    #[cfg_attr(feature = "cargo-clippy", allow(not_unsafe_ptr_arg_deref))]
+    #[allow(clippy::not_unsafe_ptr_arg_deref)]
```

Comment:
We should probably remove this function? It says backwards compatibility :)

Reply:
Yes, it not used